src/mame/drivers/ms0515.cpp

```diff
@@ -1,44 +1,119 @@
 // license:BSD-3-Clause
-// copyright-holders:Miodrag Milanovic
+// copyright-holders:Miodrag Milanovic, Sergey Svishchev
 /***************************************************************************
 
-        Elektronika MS-0515
+	Elektronika MS-0515
 
-        06/08/2012 Skeleton driver.
+	To do:
+	- softlist
+	- sound
+	- 512K memory expansion
+	- ?? refresh rate change
+	- ?? parallel printer
+	- ?? cassette (only with Version A firmware)
+	- ?? port 177770
+
+	Docs:
+	- http://www.tis.kz/docs/MC-0515/mc0515-ed.rar schematics etc.
+	- http://www.tis.kz/docs/MC-0515/mc0515-to.rar user manual
+	- http://www.tis.kz/docs/MC-0515/hc4-to.rar technical manual
+	- http://www.tis.kz/docs/MC-0515/mc0515-po.rar diag manual
+	- http://www.tis.kz/docs/MC-0515/mc0515-osa.rar OS manual
 
 ****************************************************************************/
 
 #include "emu.h"
+
+#include "bus/rs232/rs232.h"
 #include "cpu/t11/t11.h"
-#include "machine/wd_fdc.h"
-#include "machine/ram.h"
+#include "formats/ms0515_dsk.h"
+#include "machine/clock.h"
+#include "machine/i8251.h"
 #include "machine/i8255.h"
+#include "machine/ms7004.h"
+#include "machine/pit8253.h"
+#include "machine/ram.h"
+#include "machine/wd_fdc.h"
+#include "sound/speaker.h"
+#include "sound/wave.h"
+
+#include "ms0515.lh"
+
+
+#define VERBOSE_DBG 1       /* general debug messages */
+
+#define DBG_LOG(N,M,A) \
```

Comment:
Please make this a variadic macro so it doesn't require extra parentheses around the arguments to logerror.  Look at src/emu/logmacro.h for an example.